python/cudf/cudf/core/dataframe.py

```diff
@@ -7478,12 +7478,32 @@ def cov(self, **kwargs):
         df.columns = self.columns
         return df
 
-    def corr(self):
+    def corr(self, method="pearson"):
         """Compute the correlation matrix of a DataFrame.
+
+        Parameters
+        ----------
+        method : str, Default 'pearson'
+            The correlation method to use, one of 'pearson' or 'spearman'.
+
+        Returns
+        -------
+        DataFrame
+            The requested correlation matrix.
         """
-        corr = cupy.corrcoef(self.values, rowvar=False)
-        df = DataFrame(cupy.asfortranarray(corr)).set_index(self.columns)
-        df.columns = self.columns
+        if method == "pearson":
+            corr = cupy.corrcoef(self.values, rowvar=False)
+            df = DataFrame(cupy.asfortranarray(corr)).set_index(self.columns)
```

Comment:
`.columns` API are external-only API (see https://github.com/rapidsai/cudf/pull/10263). `DataFrame` is not external-only but we should opt for explicit construction whenever possible. Here, I would suggest building the index and columns explicitly and construct the dataframe with `_from_data`.

```suggestion
            column_names = cudf.core.column.as_column(self._column_names)
            index = cudf.core.index._index_from_data({None: column_names})
            df = DataFrame._from_data({col: cudf.core.column.as_column(arr) for col, arr in zip(self._column_names, corr)}, index=index)
```

Reply:
Instead of the suggestion above, you should apply this change. `Dataframe.columns` is a reserved attribute only for external usage due to it's overhead on specific return types to match pandas behavior.
```suggestion
            df = DataFrame(cupy.asfortranarray(corr)).set_index(self._columns_names)
```